src/test/scala/org/scanet/models/RegressionSpec.scala

```diff
@@ -24,7 +24,7 @@ class RegressionSpec extends AnyFlatSpec with CustomMatchers {
     val ds = CSVDataset("linear_function_1.scv")
     val weights = Optimizer
       .minimize(Regression.linear)
-      .using(SGD(momentum = 0.9, nesterov = true))
```

Comment:
Could you make a separate test for `AdaDelta`, so we would have a regression test for every alg?